packages/core/src/features/css-class.ts

```diff
@@ -0,0 +1,198 @@
+    },
+};
+
+// HOOKS
+
+export const hooks = createFeature({
+    analyzeInit({ data }) {
+        plugableRecord.set(data, dataKey, {});
+    },
+    analyzeSelectorNode<AST extends ImmutableClass>(
+        meta: StylableMeta,
+        node: AST,
+        rule: postcss.Rule
+    ): void {
+        if (node.nodes) {
+            // error on functional class
+            meta.diagnostics.error(
+                rule,
+                diagnostics.INVALID_FUNCTIONAL_SELECTOR(`.` + node.value, `class`),
+                { word: stringifySelector(node) }
+            );
+        }
+        addClass(meta, node.value, rule);
+    },
+    transformSelectorNode<AST extends Class>(context: Required<ScopeContext>, node: AST): void {
+        const { originMeta, resolver } = context;
+        const resolved = context.metaParts.class[node.value] || [
+            // used to namespace classes from js mixins since js mixins
+            // are scoped in the context of the mixed-in stylesheet
+            // which might not have a definition for the mixed-in class
+            { _kind: 'css', meta: originMeta, symbol: { _kind: 'class', name: node.value } },
+        ];
+        context.setCurrentAnchor({ name: node.value, type: 'class', resolved });
+        const { symbol, meta } = getOriginDefinition(resolved);
+        if (context.originMeta === meta && symbol[valueMapping.states]) {
+            // ToDo: refactor out to transformer validation phase
+            validateRuleStateDefinition(context.rule, meta, resolver, originMeta.diagnostics);
+        }
+        namespaceClass(meta, symbol, node, originMeta);
+    },
+});
+
+// API
+
+export function getClass({ data }: StylableMeta, name: string): ClassSymbol | undefined {
+    const state = plugableRecord.getUnsafe(data, dataKey);
+    return state[name];
+}
+
+export function getSymbols({ data }: StylableMeta): Record<string, ClassSymbol> {
+    return plugableRecord.getUnsafe(data, dataKey);
+}
+
+export function addClass(meta: StylableMeta, name: string, rule?: postcss.Rule): ClassSymbol {
+    const cssClassData = plugableRecord.getUnsafe(meta.data, dataKey);
+    let classSymbol = cssClassData[name];
+    if (!classSymbol) {
+        let alias = STSymbol.getSymbol(meta, name);
+        if (alias && alias._kind !== 'import') {
+            alias = undefined;
+        }
+        classSymbol = cssClassData[name] = {
+            _kind: 'class',
+            name,
+            alias,
+        };
+        STSymbol.addSymbol({
+            meta,
+            symbol: classSymbol,
+            node: rule,
+            safeRedeclare: !!alias,
+        });
+        // deprecated
+        ignoreDeprecationWarn(() => {
+            meta.classes[name] = classSymbol;
+        });
+    }
+    return classSymbol;
+}
+
+export function namespaceClass(
+    meta: StylableMeta,
+    symbol: StylableSymbol,
+    node: FunctionalSelector,
```

Comment:
this is called for to transform 3 types: `class`, imported root `type-element` and `pseudo_element`, non are "functional" and the namespace class will get rid of the parentheses for `-st-global` in the process by converting them into `selector` types if they are anyway.

Changed the argument type to `SelectorNode` and the convert to `convertToSelector` instead of `flattenFunctionalSelector`.